src/NzbDrone.Core/MediaFiles/DiskScanService.cs

```diff
@@ -67,26 +67,29 @@ public void Scan(Series series)
         {
             var rootFolder = _diskProvider.GetParentFolder(series.Path);
 
-            if (!_diskProvider.FolderExists(rootFolder))
+            var rootFolderExists = _diskProvider.FolderExists(rootFolder);
+            if (!rootFolderExists)
             {
                 _logger.Warn("Series' root folder ({0}) doesn't exist.", rootFolder);
-                _eventAggregator.PublishEvent(new SeriesScanSkippedEvent(series, SeriesScanSkippedReason.RootFolderDoesNotExist));
+                _eventAggregator.PublishEvent(new SeriesScanSkippedEvent(series,
+                    SeriesScanSkippedReason.RootFolderDoesNotExist));
                 return;
             }
 
-            if (_diskProvider.GetDirectories(rootFolder).Empty())
+            if (_diskProvider.GetDirectories(rootFolder).Empty() && !_configService.UpdateSeriesIfFolderMissing && !_configService.CreateEmptySeriesFolders)
```

Comment:
Since I'm on Windows and use an internal HDD it will always be mounted, so I thought it would be useful to be able to override it, but I see the point, I'll change that back.

I like the idea of removing the setting and just doing it if the folder has content since you know it's mounted. I avoided that at first because there's an edge case that if you do in fact completely empty the folder out it won't reflect that, but I think since it'll update as soon as something is in there it's a decent compromise.